tools/kl_divergence/test_local.py

```
@@ -0,0 +1,341 @@
+
+        bf = sorted(base_meta["files"])[0]
+        tf = sorted(target_meta["files"])[0]
+
+        with safe_open(os.path.join(base_dir, bf), framework="pt") as f:
+            base_tids = f.get_tensor("token_ids")
+        with safe_open(os.path.join(target_dir, tf), framework="pt") as f:
+            target_tids = f.get_tensor("token_ids")
+
+        mismatch_detected = not torch.equal(base_tids, target_tids)
+        assert mismatch_detected, "Different random token IDs should not be equal"
+        print(f"  PASSED (mismatch correctly detected)")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)